libs/pika/async_mpi/include/pika/async_mpi/transform_mpi.hpp

```diff
@@ -383,22 +389,49 @@ namespace pika::mpi::experimental {
       : pika::functional::detail::tag_fallback<transform_mpi_t>
     {
     private:
+        // if the user did not specify an mpi stream
         template <typename Sender, typename F,
             PIKA_CONCEPT_REQUIRES_(
                 pika::execution::experimental::is_sender_v<Sender>)>
         friend constexpr PIKA_FORCEINLINE auto tag_fallback_invoke(
             transform_mpi_t, Sender&& sender, F&& f)
         {
             return transform_mpi_detail::transform_mpi_sender<Sender, F>{
-                PIKA_FORWARD(Sender, sender), PIKA_FORWARD(F, f)};
+                PIKA_FORWARD(Sender, sender), PIKA_FORWARD(F, f),
+                mpi::experimental::default_stream};
         }
 
+        // if the user did specify an mpi stream
+        template <typename Sender, typename F,
+            PIKA_CONCEPT_REQUIRES_(
+                pika::execution::experimental::is_sender_v<Sender>)>
+        friend constexpr PIKA_FORCEINLINE auto tag_fallback_invoke(
+            transform_mpi_t, Sender&& sender, F&& f,
+            mpi::experimental::stream_index s)
+        {
+            return transform_mpi_detail::transform_mpi_sender<Sender, F>{
+                PIKA_FORWARD(Sender, sender), PIKA_FORWARD(F, f), s};
+        }
```

Comment:
Isn't one overload enough?
```suggestion
        template <typename Sender, typename F,
            PIKA_CONCEPT_REQUIRES_(
                pika::execution::experimental::is_sender_v<Sender>)>
        friend constexpr PIKA_FORCEINLINE auto tag_fallback_invoke(
            transform_mpi_t, Sender&& sender, F&& f,
            stream_index s = stream_index::default_)
        {
            return transform_mpi_detail::transform_mpi_sender<Sender, F>{
                PIKA_FORWARD(Sender, sender), PIKA_FORWARD(F, f), s};
        }
```